src/libstd/thread/local.rs

```diff
@@ -466,35 +565,35 @@ pub mod os {
 
     struct Value<T: 'static> {
         key: &'static Key<T>,
-        value: UnsafeCell<Option<T>>,
+        value: UnsafeCell<__LocalKeyValue<T>>,
     }
 
     impl<T: 'static> Key<T> {
         pub const fn new() -> Key<T> {
             Key {
                 os: OsStaticKey::new(Some(destroy_value::<T>)),
-                marker: marker::PhantomData
+                dummy_destroyed: UnsafeCell::new(__LocalKeyValue::Destroyed),
             }
         }
 
-        pub unsafe fn get(&'static self) -> Option<&'static UnsafeCell<Option<T>>> {
+        pub unsafe fn get(&self) -> &'static UnsafeCell<__LocalKeyValue<T>> {
             let ptr = self.os.get() as *mut Value<T>;
             if !ptr.is_null() {
                 if ptr as usize == 1 {
-                    return None
+                    // The destructor was already called (and set self.os to 1).
+                    return &*(&self.dummy_destroyed as *const _);
```

Comment:
How come `get` no longer takes `'static self` to require these casts?

Reply:
Oh, honestly I think I wasn't sure what to do there so I just copied what `mod fast` does, but of course that doesn't make sense because `mod os` uses a truly static value. I'll change it back.